addons/l10n_cl/demo/account_demo.py

```diff
@@ -1,41 +1,386 @@
+            'fiscal_position_id': fiscal_position_exempt,
+            'journal_id': local_sale_journal.id,
+            'invoice_payment_term_id': 'account.account_payment_term_end_following_month',
+            'invoice_date': today,
+            'invoice_line_ids': [
+                Command.create({'product_id': 'product.product_product_1_product_template', 'price_unit': 1230000.0, 'quantity': 1}),
+            ],
+        }
+        # credit note with a discount over product product_product_2 for invoice demo_cl_invoice_1
+        move_data['demo_cl_credit_note_discount_1'] = {
+            'company_id': company.id,
+            'move_type': 'out_refund',
+            'l10n_latam_document_type_id': 'l10n_cl.dc_nc_f_dte',
+            'partner_id': 'l10n_cl.demo_partner_cl_1',
+            'journal_id': local_sale_journal.id,
+            'invoice_payment_term_id': 'account.account_payment_term_end_following_month',
+            'invoice_date': today,
+            'invoice_line_ids': [
+                Command.create({'name': 'Discount', 'price_unit': 24200.0, 'quantity': 1}),
+            ],
+        }
+        # debit note cancelling discount credit note
+        move_data['demo_cl_debit_note_discount_1'] = {
+            'company_id': company.id,
+            'move_type': 'out_invoice',
+            'l10n_latam_document_type_id': 'l10n_cl.dc_nd_f_dte',
+            'partner_id': 'l10n_cl.demo_partner_cl_1',
+            'journal_id': local_sale_journal.id,
+            'invoice_payment_term_id': 'account.account_payment_term_end_following_month',
+            'invoice_date': today,
+            'invoice_line_ids': [
+                Command.create({'name': 'Discount', 'price_unit': 24200.0, 'quantity': 1}),
+            ],
+        }
+        # 5 receipts
+        products_for_receipts = [
+            '1', '2', '5', '7', '9', '10', '11', '16', '20', '22', '24', '27',
+        ]
+        for receipt in range(1, 5):
+            move_data[f'demo_cl_boleta_{receipt}'] = {
+                'company_id': company.id,
+                'move_type': 'out_invoice',
+                'l10n_latam_document_type_id': 'l10n_cl.dc_b_f_dte',
+                'partner_id': 'l10n_cl.par_cfa',
+                'journal_id': local_sale_journal.id,
+                'invoice_payment_term_id': 'account.account_payment_term_end_following_month',
+                'invoice_date': today - relativedelta(months=1),
+                'invoice_line_ids': [
+                    Command.create({'product_id': f'product.product_product_{products_for_receipts[receipt % len(products_for_receipts)]}', 'quantity': 1}),
+                    Command.create({'product_id': f'product.product_product_{products_for_receipts[(receipt * 2) % len(products_for_receipts)]}', 'quantity': 3}),
+                    Command.create({'product_id': f'product.product_product_{products_for_receipts[(receipt * 3) % len(products_for_receipts)]}', 'quantity': 4}),
+                ],
+            }
```

Comment:
this is the most overly-complicated-for-no-reason code I've seen in a while.

for example, this would be much more clearer:
``` python
data = [
    ('product.product_product_1',  'product.product_product_2', 'product.product_product_3'),
    ('product.product_product_2',  'product.product_product_4', 'product.product_product_6'),
    ....
]
for receipt_num, product_ids in zip(range(5), data):
```